_posts/2019-11-28-making-software-is-hard.md

```diff
@@ -0,0 +1,62 @@
+
+The real problem is that once you know that you will go over time there is not really a solution to speed up.
+
+- Crunching
+
+A lot of companies decide to force people to work overtime to try and complete a project in time. This will work for the first week, but as time goes on and people get less sleep consistently the start to perform worse and worse and you end up paying them overtime for less work than they would have done if they well working 8 hours, but well rested. This is not taking into consideration all the avverse effect on health from being overworked and not sleeping enough. Also you increase the risk of people burning out from overworking and taking productivity even further.
+
+- Hiring more people
+
+Throwing more people into the arena increases complexity so you will waste more time dealing with tooling, interpersonal relashionships, some work will be blocked by pending work from someone else and so on. It might work for certain projects, but generally you only get a 0.x increase in productivity and not the expected doubling, this factor also decreases as you add more people. If 2 people instead of one gives a 1.7 factor 3 people might give a 2.2 factor (numbers not accurate, just to give an idea).
+
+- Decreasing Scope
+
+The only real solution is a change in scope. If the deadline can't be changed and we also defined that we can increase productivity after a certain extent the only thing left is to cut on features. What is really necessary for the release? Can something be cut out? We've seen this clearly in this year's iOS 13. The OS was not ready for the new iPhone release so Apple decided to ship a version without some of the feature announced at WWDC in June and add them a while later with iOS 13.1
+
+### Time Estimation
+
+Time estimation in software will always be hard. You are trying to create something when the whole picture is not entirely there until after you've built it. That said you can use some tricks:
+
+- Increase your estimation
+
+If some parts are not clear estimate them for as long as they might take while knowing that if it turns out that it is not as bad as assumed the estimation might be reduced.
+
+- Clarify what can be clarified
+
+Some times you only need to talk with the client/designers more to get some nebulous parts clarified
+
+- Some things will always be unknown until you begin
+
+It's in the nature of software that some things won't be clear until you actually start working on them. There is nothing you can do about that. What you can do is to take that into account while estimating that feature.
+
+## Conclusions
+
+I hope I gave you some tools to give better estimates. Unfotunately you have as developer not a lot of control on the design and product analysis part of the development process that might give you a clearer picture. You just need to be aware of it and act accordingly.
```

Comment:
Unfotunately you have as developer not a lot of control on - > Unfotunately you, as developer, have not a lot of control on

Reply:
On - > in?

Reply:
mm I think on is correct there